include/rmm/mr/device/pool_memory_resource.hpp

```diff
@@ -76,7 +77,7 @@ class pool_memory_resource final : public device_memory_resource {
     if (maximum_pool_size == default_maximum_size) maximum_pool_size_ = props.totalGlobalMem;
 
     // Allocate initial block
-    stream_free_blocks_[0].insert(block_from_upstream(initial_pool_size, 0));
+    stream_free_blocks_[stream_t{}].insert(block_from_upstream(initial_pool_size, stream_t{}));
```

Comment:
I feel like this leaves something to be desired. think it would be nice if there were a way to refer explicitly to the default stream, rather than creating a default-allocated `stream_t` at multiple places in the same statement...

Providing a `rmm::default_stream()` abstraction also provides a way to enable overriding the default stream (say a user wanted the default stream to actually be a non-default CUDA stream everywhere in their application), like we do for memory resources.  For now it could just be a factory that returns `stream_t{}`.

```suggestion
    stream_free_blocks_[default_stream()].insert(block_from_upstream(initial_pool_size, default_stream()));
```

Reply:
Ok, I added `get_default_stream` and `set_default_stream` based on the equivalent for memory resource. Was this what you had in mind?

Reply:
https://github.com/rapidsai/rmm/blob/b7fa4072cee9c413f303d2101dd20c9749c5b44c/include/rmm/detail/stream.hpp#L71-L103